### PR TITLE
feat: use runtime alias strategy

### PR DESCRIPTION
**Pull request by cmoog**
The existing shell init strategy lends itself to frequent state errors as VSCode sessions start and stop. It also carries the limitation of having to restart the shell after you start the VSCode session. 

Executing the `code_connect.py` script before every call to `code` ensures that the proper socket is detected without fail. The latency drawback is nearly imperceivable, clocking in between 30-60 ms.

With this approach, we can do away with `activate.sh` and `activate.fish`... instead all we need is a simple alias like so
```fish
function code
  ~/.local/share/code-connect/code_connect.py $argv
end
```

### CODE REVIEW

**Comment by cmoog** [comment]
Obviously, this would have broader implications for the`README.md` as well as https://github.com/chvolkmann/code-connect-fish-plugin. @chvolkmann let me know if this is something you're open to merging into your mainline.

**Comment by chvolkmann** [comment]
Looks good so far! Had the plan to implement the validity check before every `code` call somewhere in the future, so I really appreciate the pull request!

Can you make sure to [enable edit permissions to the PR](https://github.com/chvolkmann/code-connect/blob/main/CONTRIBUTING.md) once it's ready? Couldn't mark the previous PR as _merged_.

File: code_connect.py [hunk 1]
```diff
@@ -83,21 +84,12 @@ def main(shell: str = None, max_idle_time: int = MAX_IDLE_TIME):
...
-            f'export VSCODE_IPC_HOOK_CLI="{ipc_sock}"',
-            f'alias code="{code_binary}"'
-        ]
-
-    print('\n'.join(source_lines))
+    args = sys.argv.copy()
+    args[0] = code_binary
+
+    os.environ["VSCODE_IPC_HOOK_CLI"] = str(ipc_sock)
+    proc = sp.run(args)
```
Comment (chvolkmann) [inline code_connect.py:91]: Really clean, I like it.
Does this forward the actual output of the `code_binary` to stdin/stderr, i.e. does it _not_ capture it some way?

Comment (cmoog) [reply code_connect.py:91]: Yes, stdout/stderr of this child process is connected to the `code_connect.py` process. Example:
```
╰─>$ ./code_connect.py
code 1.53.0

Usage: code [options][paths...]

To read from stdin, append '-' (e.g. 'ps aux | grep code | code -')

...
```

**Comment by cmoog** [comment]
Hopefully the permissions issue is fixed now.
Screen Shot 2021-02-14 at 5 16 53 PM

**Comment by chvolkmann** [comment]
I've unclutted `main()` a bit and updated the README's usage instructions. Both bash and fish support

    alias code="/path/to/code_connect.py"

Thanks again for your contribution, I really appreciate it 🥳

**Pull request merged**